src/pages/Control.jsx

```diff
@@ -11,50 +11,63 @@ import ErrorMessage from "../molecules/ErrorMessage";
 import GlobalState from "../GlobalState";
 import LoadingIndicator from "../atoms/LoadingIndicator";
 
-import {
-  mockControlData,
-  mockNextControlId,
-  mockInheritedComponentNarratives,
-} from "../mockData/ControlPage";
-
 const ERROR_MESSAGE = "Error loading project control";
 
 export default function Control() {
-  const { id } = useParams();
-
+  const { id, controlId } = useParams();
   const [state, setState] = useContext(GlobalState);
-  const [hasError, setHasError] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
+  const [control, setControl] = useState();
+  const [componentData, setComponentData] = useState();
 
   useEffect(() => {
     if (!state.project || state.project.id !== parseInt(id)) {
       setIsLoading(true);
       RequestService.get(
-        `${Config("backendUrl")}/projects/${id}/`,
+        `${Config("backendUrl")}/projects/${id}/controls/${controlId}/`,
         (response) => {
           setState((state) => ({ ...state, project: response.data }));
+          let control = response.data.catalog_data;
```

Comment:
Similarly to other comments, if you break out this logic into testable methods we gain self-documenting code and unit tests.

Reply:
This is setting the data from the API, into the react components. We have unit tests on all the data being used, and integration tests are what make sure that the API and UI are communicating correctly together.